### PR TITLE
[FL-2864] NFC update detect reader

### PR DESCRIPTION
**Pull request by gornekich**
# What's new

- Make mfkey32 nonces collection more interactive
- Limit total nonces collection to 10

# Verification 

- Detect reader works as described in Miro

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/main/nfc/scenes/nfc_scene_detect_reader.c [hunk 1]
```diff
@@ -20,30 +22,56 @@ void nfc_scene_detect_reader_on_enter(void* context) {
...
         } else if(event.event == NfcWorkerEventDetectReaderMfkeyCollected) {
-            detect_reader_inc_nonce_cnt(nfc->detect_reader);
+            nonces_collected += 2;
+            scene_manager_set_scene_state(
+                nfc->scene_manager, NfcSceneDetectReader, nonces_collected);
+            detect_reader_set_nonces_collected(nfc->detect_reader, nonces_collected);
+            if(nonces_collected >= NFC_SCENE_DETECT_READER_PAIR_NONCES_MAX) {
+                detect_reader_set_state(nfc->detect_reader, DetectReaderStateDone);
+                nfc_blink_stop(nfc);
+                notification_message(nfc->notifications, &sequence_success);
```
Comment (Astrrra) [inline applications/main/nfc/scenes/nfc_scene_detect_reader.c]: The LED just blinks, it doesn't stay glowing

File: applications/main/nfc/scenes/nfc_scene_detect_reader.c [hunk 2]
```diff
@@ -20,30 +22,56 @@ void nfc_scene_detect_reader_on_enter(void* context) {
...
+                nfc_worker_stop(nfc->worker);
+            }
             consumed = true;
+        } else if(event.event == NfcWorkerEventDetectReaderDetected) {
+            if(nonces_collected < NFC_SCENE_DETECT_READER_PAIR_NONCES_MAX) {
+                nfc_blink_stop(nfc);
+                notification_message(nfc->notifications, &sequence_blink_start_magenta);
```
Comment (Astrrra) [inline applications/main/nfc/scenes/nfc_scene_detect_reader.c]: The board says that should be cyan, not magenta

**Comment by Astrrra** [comment]
Btw, should we show all the saved nonce pairs after a successful read? I think that only showing the new ones would be better, as you won't have to scroll through possibly hundreds of previous ones

File: applications/main/nfc/scenes/nfc_scene_mfkey_nonces_info.c [hunk 3]
```diff
@@ -16,14 +16,14 @@ void nfc_scene_mfkey_nonces_info_on_enter(void* context) {
 
     uint16_t nonces_saved = mfkey32_get_auth_sectors(temp_str);
     widget_add_text_scroll_element(nfc->widget, 0, 22, 128, 42, string_get_cstr(temp_str));
-    string_printf(temp_str, "Nonces saved %d", nonces_saved);
+    string_printf(temp_str, "Nonce pairs saved %d", nonces_saved);
```
Comment (Astrrra) [inline applications/main/nfc/scenes/nfc_scene_mfkey_nonces_info.c]: I think that "Saved %d nonce pairs" would fit better here

**Review by Astrrra — DISMISSED** [review]

File: applications/main/nfc/scenes/nfc_scene_mfkey_nonces_info.c (hunk 3 above)
Comment (gornekich) [reply applications/main/nfc/scenes/nfc_scene_mfkey_nonces_info.c]: Personally I prefer "description - value" format. Also it complies with Miro

**Comment by gornekich** [comment]
It's hard to separate new nonces from old since all mfkey parameters are saved in file the same moment as we get them

**Review by skotopes — DISMISSED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**